src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java

```diff
@@ -231,6 +266,22 @@ SELECT COUNT (DISTINCT p) FROM ProgrammingExerciseStudentParticipation p join p.
             """)
     long countSubmissionsByCourseIdSubmitted(@Param("courseId") Long courseId);
 
+    /**
+     * In distinction to other exercise types, students can have multiple submissions in a programming exercise.
+     * We therefore have to check here if any submission of the student was submitted before the deadline.
+     *
+     * @param exerciseIds the course id we are interested in
+     * @return the number of submissions belonging to the course id, which have the submitted flag set to true and the submission date before the exercise due date, or no exercise
+     *         due date at all (only exercises with manual or semi automatic correction are considered)
```

Comment:
This does not appear to check any due dates.

Reply:
I fixed the comment